base/abstractarray.jl

```diff
@@ -1047,6 +1047,88 @@ function _setindex!(::IndexCartesian, A::AbstractArray, v, I::Vararg{Int,M}) whe
     r
 end
 
+"""
+    parent(A)
+
+Returns the "parent array" of an array view type (e.g., `SubArray`), or the array itself if
+it is not a view.
+
+# Examples
+```jldoctest
+julia> A = [1 2; 3 4]
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+
+julia> V = view(A, 1:2, :)
+2×2 view(::Array{Int64,2}, 1:2, :) with eltype Int64:
+ 1  2
+ 3  4
+
+julia> parent(V)
+2×2 Array{Int64,2}:
+ 1  2
+ 3  4
+```
+"""
+parent(a::AbstractArray) = a
+
+## rudimentary aliasing detection ##
+"""
+    unalias(dest, A)
+
+Return either `A` or a copy of `A`, in a rough effort to prevent modifications to `dest` from
+affecting the returned object. No guarantees are provided, and each custom array must
+opt-into aliasing detection by overloading this method by specializing on the second argument.
+
+This function must return an object of exactly the same type as `A` for performance and type stability.
+
+See also [`mightalias`](@ref) and [`dataids`](@ref).
+"""
+unalias(dest, A::AbstractArray) = mightalias(dest, A) ? copypreservingtype(A) : A
+unalias(dest, A::AbstractRange) = A
+unalias(dest, A) = A
+
+copypreservingtype(A::Array) = copy(A)
```

Comment:
This sounds like "copy-preserving type" and too similar to `eltype()`, `keytype()` etc.
Maybe something along `sametypecopy()`/`keeptypecopy()`/`unaliasingcopy()` (my fav)?

Reply:
Fair point.  I also like `unaliasingcopy` for now — it's likely that we'll try to separate the two meanings of copy in the future.  By naming this more narrowly it gives us space for that design process.